### PR TITLE
support customizable loss scale

### PR DESCRIPTION
**Pull request by hjh0119**
# PR type
- [ ] Bug Fix
- [x] New Feature
- [ ] Document Updates
- [x] More Models or Datasets Support

Customizable configuration of loss_scale is supported, including the matching of response and query fields. A default config is also provided.
Support msagent-pro dataset.
Support toolbench dataset

### CODE REVIEW

File: swift/llm/utils/argument.py [hunk 1]
```diff
@@ -460,6 +460,7 @@ class SftArguments(ArgumentsBase):
     dataset_seed: int = 42
     dataset_test_ratio: float = 0.01
     use_loss_scale: bool = False  # for agent
+    loss_scale_config_path: Optional[str] = 'DEFAULT'
```
Comment (Jintao-Huang) [inline swift/llm/utils/argument.py]: Optional. 有None的时候么

File: swift/llm/agent/default_loss_scale_config.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (Jintao-Huang) [inline swift/llm/agent/default_loss_scale_config.json:1]: 这个要入包嘛, 如果要入的话, 更新一下MANIFEST.in文件

File: swift/llm/agent/utils.py [hunk 3]
```diff
@@ -90,41 +93,25 @@ def calculate_loss_scale(response: str, use_loss_scale=False) -> Tuple[List[str]
     Returns:
         A tuple of agent response parts and their weights.
     """
-    if 'Action:' in response and 'Observation:' in response and use_loss_scale:
-        agent_keyword = ['Action:', 'Action Input:', 'Thought:', 'Final Answer:', 'Observation:']
-        agent_parts = split_str_parts_by(response, agent_keyword)
+    if use_loss_scale:
+        if loss_scale_value:
+            return [response], [float(loss_scale_value)]
```
Comment (Jintao-Huang) [inline swift/llm/agent/utils.py]: loss_scale_value我看是Optional[Union[str, float, int]], 如果是字符串是不是会报错. float(...)

File: swift/llm/utils/template.py [hunk 4]
```diff
@@ -279,6 +288,7 @@ def _concat_context_list(
         query: Optional[str] = None,
         response: Optional[str] = None,
         round0: Optional[int] = None,
+        loss_scale_value: Optional[Union[str, float, int]] = None,
```
Comment (Jintao-Huang) [inline swift/llm/utils/template.py]: str检查一下, 我不确定

**Review by tastelikefeet — APPROVED** [review]

**Pull request merged**